src/sql/src/plan/statement/ddl.rs

```diff
@@ -2143,9 +2144,13 @@ generate_extracted_config!(CreateSinkOption, (Size, String), (Snapshot, bool));
 
 pub fn plan_create_sink(
     scx: &StatementContext,
-    stmt: CreateSinkStatement<Aug>,
+    mut stmt: CreateSinkStatement<Aug>,
 ) -> Result<Plan, PlanError> {
+    // updating avro format with comments so that they are frozen in the `create_sql`
+    update_avro_format_with_comments(scx, &mut stmt)?;
```

Comment:
I think this logic belongs in `purify_statement`, then it doesn't need to be mut here.

Reply:
Ah, let me update.

Reply:
I just looked at the code. The `purify_statement` is resolving external dependencies and returns vector with `CreateSubsourceStatement`, which is not applicable here since there are no external dependencies and no subsources. I could return an empty vector, but it does not seem well suited?

Reply:
Agree; my mistake. I hadn't looked at that function deeply and didn't realize it's only called in one place. LGTM.